crypto/asn1/a_strnid.c

```diff
@@ -129,8 +129,10 @@ ASN1_STRING_TABLE *ASN1_STRING_TABLE_get(int nid)
     int idx;
     ASN1_STRING_TABLE fnd;
 
+#ifndef OPENSSL_NO_AUTOLOAD_CONFIG
     /* "stable" can be impacted by config, so load the config file first */
     OPENSSL_init_crypto(OPENSSL_INIT_LOAD_CONFIG, NULL);
```

Comment:
Another way of doing this would be to put the code into OPENSSL_init_crypto instead..
````
#ifdef OPENSSL_NO_AUTOLOAD_CONFIG
 if (opts == OPENSSL_INIT_LOAD_CONFIG)
      return 1;
#endif

Reply:
That would be future proof..

Reply:
No it wouldn't.  Calling OPENSSL_init_crypto with the OPENSSL_INIT_LOAD_CONFIG flag is perfectly legitimate from an application even with this compile time option.  What mustn't happen is the library ever making such a call.